BUILD.md

```diff
@@ -58,7 +58,23 @@ The complete list of build options can be found by running `./build.sh (or ./bui
 
 ## Build ONNX Runtime Server on Linux
 
-1. In the ONNX Runtime root folder, run `./build.sh --config RelWithDebInfo --build_server  --use_openmp --parallel`
+1. Install Protobuf and GRPC from source code. 
+    1. Go to cmake/external/protobuf
+    2. ```
+       ./autogen.sh
+       ./configure --disable-shared
+       make check
+       sudo make install
+       sudo ldconfig
+       ```
+    3. Go to cmake/external/grpc
+    4. ```
```

Comment:
The problem is: In the cmake/external/grpc directory, it contains another version of protobuf. That's the one grpc wants. If you replace it with the one in onnxruntime's directory, by coincidence it may work, but it's not good.  Because it will make onnxruntime indirectly depends on grpc. onnxruntime must pick a protobuf that also works fine with gprc.  I don't want to see that connection.

Reply:
Is the concern with having another version checked into the source, or is the concern about onnx RT server constraining the version of libprotobuf in core? 

OnnxRuntimeServer needs ABI-compatible versions of the ONNX_NAMESPACE::TensorProto so it can use utils::TensorProtoToMLValue, so the protobuf library versions can't go out of sync anyways - dynamic linking would only complicate things. 

My understanding is GRPC is pretty broadly compatible with protobuf because the core library is built in a serialization unaware fashion. The codegen plugin would depend on protoc, and the generated code would depend on libprotobuf, but those should be fairly stable.

Reply:
When talking ABI, TensorProto is absolutely out of the scope.  You can't make TensorProto  ABI compatible, because it contains C++ symbols, and there no ABI for C++ (at least on Windows).

The problem with TensorProtoToMLValue is OnnxRuntimeServer's own business. onnxruntime do not accept TensorProto from application. OnnxRuntimeServer is passing around the TensorProto within its own code, it won't pass such an object to onnxruntime engine.  So, I would suggest let decouple the things. If OnnxRuntimeServer dynamically link to OnnxRuntime, then OnnxRuntimeServer can **independently** make the decision of which third-party library and which version of the library it wants to use.

Reply:
>OnnxRuntimeServer needs ABI-compatible versions of the ONNX_NAMESPACE::TensorProto so it can use utils::TensorProtoToMLValue

@csteegz you may not be aware but the eventual goal is to completely decouple server from the rest